### PR TITLE
teng's coffee style guide

### PR DESCRIPTION
**Pull request by tengbao**
1. allow block comments using ###
2. discourage weird function grouping by parens
3. prefer `&&` over `and`, etc
4. enforce explicit `return` for multiline functions
5. prefer no spaces in function def default params
6. emphasize `when` in for loops

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -435,20 +423,13 @@ If a custom annotation is required, the annotation should be documented in the p
...
 
-`and` is preferred over `&&`.
-
-`or` is preferred over `||`.
-
-`is` is preferred over `==`.
+`&&` is preferred over `and`.
 
-`not` is preferred over `!`.
+`||` is preferred over `or`.
```
Comment (rechtar) [inline README.md]: can note that the reason here is to minimize the mental gap between frontend code and ruby

File: README.md [hunk 2]
```diff
@@ -435,20 +423,13 @@ If a custom annotation is required, the annotation should be documented in the p
...
 
-`or=` should be used when possible:
+`==` is preferred over `is`.
```
Comment (syjstc) [inline README.md]: i think 'is' can help us always think about the type of variables

Comment (rechtar) [inline README.md]: `is` doesn't mean the same thing in ruby and i think that makes a good reason we don't use it - same with other operators

Comment (dfguo) [inline README.md]: `is` doesn't exist in ruby

File: README.md [hunk 3]
```diff
@@ -465,13 +448,26 @@ return this.property # No
...
 
 However, avoid the use of **standalone** `@`:
-
 ```coffeescript
 return this # Yes
 return @ # No
 ```
 
-Avoid `return` where not required, unless the explicit return increases clarity.
+Always use explicit `return` in multi-line functions to improve clarity.
```
Comment (dfguo) [inline README.md:469]: I'm not sure about this. Why?

**Comment by dfguo** [comment]
reviewed

**Pull request merged**